File: compare.py
from flask import Blueprint, request, jsonify
from services.llm import compare_clauses

compare_bp = Blueprint("compare_bp", __name__)

@compare_bp.route("/compare", methods=["POST"])
def compare():
    data = request.json
    a = data.get("clauseA")
    b = data.get("clauseB")

    result = compare_clauses(a, b)
    return jsonify({"result": result})

File: compare.js
async function compareClauses() {
    let a = document.getElementById("clauseA").value;
    let b = document.getElementById("clauseB").value;

    if (!a || !b) {
        alert("Enter both clauses.");
        return;
    }

    document.getElementById("compareResult").innerText = "Comparing…";

    let res = await fetch("/api/compare", {
        method: "POST",
        headers: {"Content-Type": "application/json"},
        body: JSON.stringify({
            clauseA: a,
            clauseB: b
        })
    });

    let data = await res.json();

    document.getElementById("compareResult").innerText = data.result;
}
